main.py

```diff
@@ -0,0 +1,58 @@
+import logging
+import os
+
+import yaml
+from databricks.connect import DatabricksSession
+
+from taxinyc.config import ProjectConfig
+from taxinyc.data_processor import DataProcessor
+
+logging.basicConfig(level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s")
+logger = logging.getLogger(__name__)
+
+try:
+    config = ProjectConfig.from_yaml(config_path="./project_config.yml")
+    logger.info("Configuration loaded successfully")
+except FileNotFoundError:
+    logger.error("Configuration file 'project_config.yml' not found")
+    raise
+except yaml.YAMLError as e:
+    logger.error("Error parsing configuration file: %s", e)
+    raise
+
+spark = DatabricksSession.builder.profile(os.environ["DATABRICKS_PROFILE"]).getOrCreate()
+
+try:
+    path = (
+        config.read_from["catalog_name"] + "." + config.read_from["schema_name"] + "." + config.read_from["table_name"]
```

Comment:
better to use f-string, like f"{config.read_from["catalog_name"]}.{config.read_from["schema_name"]}.{config.read_from["table_name"]}"

Reply:
I fixed that